service/statistics/AnalyticsEvents.js

```diff
@@ -1,447 +1,339 @@
- * properties with names that are already used by permanent properties
- * (unless the intention is to override a permanent property). Here is a
- * (non-exhaustive) list of currently know permanent properties:
- *
- * abtestSuspendVideo
- * browserName
- * callstatsname
- * crossRegion
- * forceJvb121
- * region
- * roomName
- * shard
- * size
- * userAgent
- * userRegion
- *
- * The naming convention for the constants below uses "_" as a prefix or
- * suffix to indicate that known usage of the constant prepends or appends
- * a string to the name of the event.
+ * The constant which identifies an event of type "operational".
+ * @type {string}
  */
+export const TYPE_OPERATIONAL = 'operational';
 
 /**
- * Properties: value
- *
- * TODO: document, reformat
- *
- * Full event names (uncertain):
- * conference.muc.joined (???)
- * conference.sharingDesktop.start (???)
- * conference.sharingDesktop.stop (???)
- * xmpp.attached (???)
- * xmpp.attaching (???)
- * xmpp.connected (???)
- * xmpp.connecting (???)
- * xmpp.session-initiate (???)
+ * The constant which identifies an event of type "page".
+ * @type {string}
  */
-export const _CONNECTION_TIMES_ = '';
+export const TYPE_PAGE = 'page';
 
 /**
- * TODO: document, reformat (group together with other ICE events)
- *
- * Known full event names:
- * ice.initiator.checksDuration
- * ice.responder.checksDuration
- * p2p.ice.initiator.checksDuration
- * p2p.ice.responder.checksDuration
- *
- * Properties: value
+ * The constant which identifies an event of type "track".
+ * @type {string}
  */
-export const _ICE_CHECKING_DURATION = 'checksDuration';
+export const TYPE_TRACK = 'track';
 
 /**
- * TODO: document, reformat
- *
- * Known full event names:
- * ice.checking
- * ice.closed
- * ice.completed
- * ice.connected
- * ice.disconnected
- * ice.failed
- * p2p.ice.checking
- * p2p.ice.closed
- * p2p.ice.completed
- * p2p.ice.connected
- * p2p.ice.disconnected
- * p2p.ice.failed
- *
- * Properties: value
+ * The constant which identifies an event of type "ui".
+ * @type {string}
  */
-export const _ICE_CONNECTION_STATE_ = 'ice';
+export const TYPE_UI = 'ui';
 
 /**
- * TODO: document, reformat (group together with other ICE events)
- *
- * Known full event names:
- * ice.initiator.establishmentDuration
- * ice.responder.establishmentDuration
- * p2p.ice.initiator.establishmentDuration
- * p2p.ice.responder.establishmentDuration
- *
- * Properties: value
+ * Events with source 'sporadic' are sent from various places in the code, and
+ * indicate things which happen any action from the user. These events do not
+ * happen in every conference, and do not happen periodically. They are grouped
+ * together by this value for the 'source' attribute, because they may be of
+ * interest when looking for events in a particular conference.
+ * @type {string}
  */
-export const _ICE_ESTABLISHMENT_DURATION = 'establishmentDuration';
+const SOURCE_SPORADIC = 'sporadic';
```

Comment:
I'm wondering if it's worthwhile to make separate files for event names and event attributes/types and maybe event factories. What do you think?

Reply:
I'm not sure. It would be a little bit harder to find the code given a resulting event. And importing is more of a hassle. I'd keep it in one file.